konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/api/context/TraceEvent.java

```diff
@@ -0,0 +1,19 @@
+package ai.konduit.serving.pipeline.api.context;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class TraceEvent {
+    public static enum EventType {
+        START, END;
+    }
+    private String name;
+    private long timeStamp;
+    EventType type;
```

Comment:
If we're using TraceEvent to directly write JSON, the format (and hence fields names) need to match the chrome tracing format: "name", "ts", "ph".
As it stands, the JSON produced by serializing this class won't be readable.

https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit

We also need process ID, thread ID, and category.
See also TraceEvent class in ND4J (for SameDiff).

Alternatively for JSON serialization we can use a  custom JSON serializer - see DataJsonSerializer and the JsonSerialize/Deserialize on Data interface for an example on this. But simply naming the fields to match the required JSON format is easier IMO.